### PR TITLE
busybox -> 1.33.1

### PR DESCRIPTION
**Pull request by saltedcoffii**
Upgrade busybox. This needs binaries. Also UPX doesn't seem to want to compress it.

### CODE REVIEW

File: packages/busybox.rb [hunk 1]
```diff
@@ -14,1200 +14,1205 @@ class Busybox < Package
...
+    CONFIG_SHOW_USAGE=y
+    CONFIG_FEATURE_VERBOSE_USAGE=y
+    CONFIG_FEATURE_COMPRESS_USAGE=y
+    CONFIG_LFS=y
+    # CONFIG_PAM is not set
+    CONFIG_FEATURE_DEVPTS=y
+    # CONFIG_FEATURE_UTMP is not set
+    # CONFIG_FEATURE_WTMP is not set
+    # CONFIG_FEATURE_PIDFILE is not set
+    CONFIG_PID_FILE_PATH="/usr/local/var/run"
```
Comment (uberhacker) [inline packages/busybox.rb:41]: This should be abstracted with `#{CREW_PREFIX}` in place of `/usr/local`.

File: packages/busybox.rb [hunk 2]
```diff
@@ -14,1200 +14,1205 @@ class Busybox < Package
...
+    # Installation Options ("make install" behavior)
+    #
+    CONFIG_INSTALL_APPLET_SYMLINKS=y
+    # CONFIG_INSTALL_APPLET_HARDLINKS is not set
+    # CONFIG_INSTALL_APPLET_SCRIPT_WRAPPERS is not set
+    # CONFIG_INSTALL_APPLET_DONT is not set
+    # CONFIG_INSTALL_SH_APPLET_SYMLINK is not set
+    # CONFIG_INSTALL_SH_APPLET_HARDLINK is not set
+    # CONFIG_INSTALL_SH_APPLET_SCRIPT_WRAPPER is not set
+    CONFIG_PREFIX="/usr/local/tmp/crew/dest/usr/local"
```
Comment (uberhacker) [inline packages/busybox.rb:85]: This should be abstracted with `#{CREW_PREFIX}` in place of `/usr/local/tmp/crew/dest/usr/local`.

File: packages/busybox.rb [hunk 3]
```diff
@@ -14,1200 +14,1205 @@ class Busybox < Package
...
+    CONFIG_FEATURE_CHAT_SWALLOW_OPTS=y
+    CONFIG_FEATURE_CHAT_SEND_ESCAPES=y
+    CONFIG_FEATURE_CHAT_VAR_ABORT_LEN=y
+    CONFIG_FEATURE_CHAT_CLR_ABORT=y
+    # CONFIG_CONSPY is not set
+    CONFIG_CROND=y
+    CONFIG_FEATURE_CROND_D=y
+    CONFIG_FEATURE_CROND_CALL_SENDMAIL=y
+    CONFIG_FEATURE_CROND_SPECIAL_TIMES=y
+    CONFIG_FEATURE_CROND_DIR="/usr/local/var/spool/cron"
```
Comment (uberhacker) [inline packages/busybox.rb:782]: This should be abstracted with `#{CREW_PREFIX}` in place of `/usr/local`.

File: packages/busybox.rb [hunk 4]
```diff
@@ -14,1200 +14,1205 @@ class Busybox < Package
...
+    CONFIG_IPADDR=y
+    CONFIG_IPLINK=y
+    CONFIG_IPROUTE=y
+    CONFIG_IPTUNNEL=y
+    CONFIG_IPRULE=y
+    CONFIG_IPNEIGH=y
+    CONFIG_FEATURE_IP_ADDRESS=y
+    CONFIG_FEATURE_IP_LINK=y
+    CONFIG_FEATURE_IP_ROUTE=y
+    CONFIG_FEATURE_IP_ROUTE_DIR="/usr/local/etc/iproute2"
```
Comment (uberhacker) [inline packages/busybox.rb:927]: This should be abstracted with `#{CREW_PREFIX}` in place of `/usr/local`.

File: packages/busybox.rb [hunk 5]
```diff
@@ -14,1200 +14,1205 @@ class Busybox < Package
...
+    CONFIG_FEATURE_IFCONFIG_STATUS=y
+    CONFIG_FEATURE_IFCONFIG_SLIP=y
+    CONFIG_FEATURE_IFCONFIG_MEMSTART_IOADDR_IRQ=y
+    CONFIG_FEATURE_IFCONFIG_HW=y
+    CONFIG_FEATURE_IFCONFIG_BROADCAST_PLUS=y
+    CONFIG_IFENSLAVE=y
+    CONFIG_IFPLUGD=y
+    CONFIG_IFUP=y
+    CONFIG_IFDOWN=y
+    CONFIG_IFUPDOWN_IFSTATE_PATH="/usr/local/var/lib/ifstate"
```
Comment (uberhacker) [inline packages/busybox.rb:904]: This should be abstracted with `#{CREW_PREFIX}` in place of `/usr/local`.

Comment (saltedcoffii) [reply packages/busybox.rb:904]: yk what, let's just `sed -i "s:/usr/local:#{CREW_PREFIX}:g" busybox.rb`

**Comment by supechicken** [comment]
Bump version?

**Comment by satmandu** [comment]
Also, upx doesn't work on a lot of binaries. But that's ok.

**Comment by uberhacker** [comment]
But it's not ok after you spend hours compiling only to find out it doesn't work. :)

**Comment by saltedcoffii** [comment]
> Bump version?

Shouldn't be necessary, busybox was already patched a version upstream.

**Comment by saltedcoffii** [comment]
> But it's not ok after you spend hours compiling only to find out it doesn't work. :)

What happened? Can I see an output log?

**Comment by uberhacker** [comment]
> > Bump version?
> 
> Shouldn't be necessary, busybox was already patched a version upstream.

I think @supechicken means the source should be https://busybox.net/downloads/busybox-1.33.1.tar.bz2.

**Comment by uberhacker** [comment]
> > But it's not ok after you spend hours compiling only to find out it doesn't work. :)
> 
> What happened? Can I see an output log?

It wasn't this PR but on builds, in general.

**Comment by satmandu** [comment]
> But it's not ok after you spend hours compiling only to find out it doesn't work. :)

Are we talking about upx?

The upx code is set to try to compress. If compression fails, it leaves the binary. If it succeeds at compression but fails at decompressing, it leaves the original binary.

Think of the disk space saved on so many end user machines if compression does succeed!

**Comment by satmandu** [comment]
building on x86_64 & armv7l...

**Comment by satmandu** [comment]
I built these before the version was updated. I will rebuild these.

File: packages/busybox.rb [hunk 6]
```diff
@@ -10,1204 +10,1220 @@ class Busybox < Package
   source_url 'https://busybox.net/downloads/busybox-1.33.0.tar.bz2'
```
Comment (uberhacker) [inline packages/busybox.rb:10]: Can we bump the version to the latest source?

**Comment by saltedcoffii** [comment]
I'm not at my computer rn, however it doesn't depend on linux_pam now

**Comment by uberhacker** [comment]
Added i686 binary.

**Pull request merged**